src/javaRosa.js

```diff
@@ -576,7 +576,12 @@ define([
 
         widget.save = function () {
             // override save to call out to rename itext
-            widget.mug.setItextId(widget.path, widget.getValue());
+            if (widget.mug.setItextId(widget.path, widget.getValue())) {
```

Comment:
Good point. I moved it to the mug. No need for having a function call in a if statement now